ext/session/session.c

```diff
@@ -1462,7 +1463,10 @@ PHPAPI zend_result php_session_reset_id(void) /* {{{ */
 	}
 
 	if (PS(use_cookies) && PS(send_cookie)) {
-		php_session_send_cookie();
+		zend_result cookies_sent = php_session_send_cookie();
```

Comment:
Just this change should suffice I think.

Reply:
I did this initially, this caused other issues that's why I did the other changes too